experiments/process_sandbox/cxxapi/cxxsandbox.h

```diff
@@ -0,0 +1,304 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include "sandbox.hh"
+/**
+ * The generic sandbox code requires that libraries provide an initialisation
+ * function and a dispatcher function, a vtable, and code for storing call
+ * frames in memory that allows the dispatcher to invoke them.  In Verona, this
+ * will be generated by `veronac`.  This file provides C++ helpers that manage
+ * everything using C++ compile-time type information.
+ *
+ * Correctly using sandboxes from C++ requires sanitising every pointer that
+ * you might follow, which C++ can't help with.  This code is therefore
+ * mainly useful for testing.
+ */
+
+namespace sandbox
+{
+  /**
+   * The argument frame for a sandboxed function.  This contains space for the
+   * return value and the arguments.
+   */
+  template<typename Ret, typename... Args>
+  struct argframe
+  {
+    /**
+     * The return value of this function.  If the return value is void, uses
+     * char as an unused placeholder.
+     */
+    std::conditional_t<std::is_void_v<Ret>, char, Ret> ret;
+    /**
+     * The arguments to the function.
+     */
+    std::tuple<std::remove_reference_t<Args>...> args;
+    // FIXME: This strips references, so we copy arguments, but should probably
+    // turn references into pointers.
```

Comment:
IIUC, the compiler will check if the pointer being passed belongs to the same cown, and if (by construction) we create all Verona objects in an area of memory that the sandbox process can see (ex. its heap), then passing pointers would always be to a visible area on both sides, correct?